Packs/MobileIronUEM/Integrations/MobileIronCLOUD/MobileIronCLOUD.py

```diff
@@ -0,0 +1,511 @@
+        outputs_key_field='id',
+        outputs=devices
+    )
+
+
+def execute_get_device_by_field_command(client: MobileIronCloudClient,
+                                        field_name: str, field_value: str) -> CommandResults:
+    """Returns a single device based on the property name matching the provided value
+
+    :type client: ``MobileIronCloudClient``
+    :param client: MobileIron UEM client to use
+
+    :type field_name: ``str``
+    :param field_name: name of the field to query for
+
+    :type field_value: ``str``
+    :param field_value: value of the field to query for
+
+    :return:
+        A ``CommandResults`` object that is then passed to ``return_results``,
+        that contains the device data
+
+    :rtype: ``CommandResults``
+    """
+    query = f'{field_name}={field_value}'
+    partition_id = get_partition_id(client)
+    devices = client.get_devices_data(partition_id=partition_id, query=query)
+    device = next(iter(devices), None)
+    return CommandResults(
+        outputs_prefix='MobileIronCloud.Device',
+        outputs_key_field='id',
+        outputs=device
+    )
+
+
+def fetch_incidents(client: MobileIronCloudClient, partition_id: str,
+                    incident_type: str, max_fetch: int) -> List[Dict[str, Any]]:
+    """This function returns incidents after analyzing the response data
+
+    This function has to implement the logic of making sure that incidents are
+    fetched based on analyzing the response data. By default it's invoked by
+    XSOAR every minute. It will use last_run to save the timestamp of the last
+    incident it processed.
+
+    :type partition_id: ``str``
+    :param partition_id: Partition ID of the tenant that contains the device.
+
+    :type incident_type: ``str``
+    :param incident_type: Incident Type to create, configured in the instance settings
+
+    :type max_fetch: ``int``
+    :param max_fetch: Cap on how many devices should be fetched from the API
+
+    :type client: ``MobileIronCloudClient``
+    :param client: MobileIron client to use
+
+    :return:
+        incidents are returned in the form of dict
+    """
+
+    incidents = []
+    devices = client.get_devices_data(partition_id=partition_id, query=FETCH_INCIDENTS_QUERY, max_fetch=max_fetch)
+
+    for device in devices:
+        # Rename keys for device attributes
+        message, severity = resolve_device_incident_severity(device)
+        incident_name = f'MobileIron Device Alert - {message}'
+
+        incident = {
+            'name': incident_name,
+            'rawJSON': json.dumps(device),
+            'type': incident_type,
+            'severity': severity
+        }
+
+        incidents.append(incident)
+
+    return incidents
+
+
+def execute_fetch_incidents_command(client):
+    last_run = demisto.getLastRun()
+    datetime_now = datetime.utcnow()
+    datetime_now_iso = datetime_now.isoformat()
+
+    params = demisto.params()
+    fetch_interval = int(params.get('fetch_interval'))
```

Comment:
We don't want the fetch query to run every minute. It will run on a configurable interval. by default 30 mins.

Reply:
if some customer wants to change their fetch they can do it via the configuration of the server.
but in this way, it makes it very not convenient for them cause it’s not like the other integrations.
we should try to keep the integration configuration close to the others.

Reply:
How do you configure this on the server? Can you point me to the documentation for that? 

as far i understood from talking with @liorkol this is not possible, and he pointed out a few other integrations that do the same thing to avoid polling every minute. Polling our API's every minute for this information is not a good approach from the use-cases/product perspective for our integration. The actions someone takes are not reflected in realtime as information needs to reach a mobile device first, there is usually a significant delay, which can be in best case scenarios 10+ mins.

I find its a much better solution if this is configurable out-of-the box on the server for the integration specifically, but it wasn't obvious for me, so it wont be for the customer using it. I would like to have it documented this and link to the related documentation for your system.

Reply:
Even in that case, we will prepare to not return any incidents but make the call.
(one call per minute is not a big thing)